src/node/blockstorage.cpp

```diff
@@ -199,18 +196,13 @@ CBlockIndex* BlockManager::InsertBlockIndex(const uint256& hash)
         return nullptr;
     }
 
-    // Return existing
-    BlockMap::iterator mi = m_block_index.find(hash);
-    if (mi != m_block_index.end()) {
-        return (*mi).second;
+    // Return existing or create new
+    auto [mi, inserted] = m_block_index.try_emplace(hash);
+    CBlockIndex* pindex = &(*mi).second;
```

Comment:
in the third commit: Any reason to use a pointer here when a reference is sufficient?

```diff
diff --git a/src/node/blockstorage.cpp b/src/node/blockstorage.cpp
index 41623186ff..cab8c20e53 100644
--- a/src/node/blockstorage.cpp
+++ b/src/node/blockstorage.cpp
@@ -203,11 +203,11 @@ CBlockIndex* BlockManager::InsertBlockIndex(const uint256& hash)
 
     // Return existing or create new
     auto [mi, inserted] = m_block_index.try_emplace(hash);
-    CBlockIndex* pindex = &(*mi).second;
+    CBlockIndex& block{(*mi).second};
     if (inserted) {
-        pindex->phashBlock = &((*mi).first);
+        block.phashBlock = &((*mi).first);
     }
-    return pindex;
+    return &block;
 }
 
 bool BlockManager::LoadBlockIndex(